ambry-rest/src/main/java/com.github.ambry.rest/AsyncRequestResponseHandler.java

```diff
@@ -317,6 +317,8 @@ public void run() {
           }
         }
       }
+    } catch (Throwable t) {
+      logger.error("AsyncRequestWorker encountered a throwable", t);
```

Comment:
Don't think there is any point in re-throwing since the thread will exit anyway (also the signature of `run()` doesn't allow `Throwable`). Along the same lines, there is no higher up here unfortunately since the thread will simply exit and there is no function that is watching for errors.

However, there might be another way to address what you bring up here - the function `newThread()` in `Utils` (which we usually use to create new threads in our code) sets an exception handler that ignores the exception. Instead of individually printing the messages, we could change that code to log the `Throwable`. Do you think that is the better approach?